### PR TITLE
Backend/feature/add is owner bookmark

### PR DESCRIPTION
**Pull request by mharasic**
Added addtional flags for is_bookmark and is_owner

### CODE REVIEW

File: backend/api/search.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def search_simple():
     count = request.args.get('count', type=int)
     if count is None:
         count = 10
+
```
Comment (henderkes) [inline backend/api/search.py]: this might let pep8 fail

File: backend/api/projects.py [hunk 2]
```diff
@@ -102,7 +102,14 @@ def get_projects():
         res = g.projects.find({}, skip=skip)
     else:
         return make_response('Invalid parameters', 400)
-
+    try:
+        for project in res:
+            project['is_bookmark'] = 'true' if str(project['id'])\
```
Comment (henderkes) [inline backend/api/projects.py]: is this '_id' maybe?

Comment (mharasic) [reply backend/api/projects.py]: ah yes
id is for ES... _is for mongo

File: backend/api/projects.py [hunk 3]
```diff
@@ -102,14 +101,23 @@ def get_projects():
         res = g.projects.find({}, skip=skip)
     else:
         return make_response('Invalid parameters', 400)
+
     try:
+        res = [x for x in res[:]]
```
Comment (henderkes) [inline backend/api/projects.py]: res = res[:] would do the same, no?

File: backend/api/projects.py [hunk 4]
```diff
@@ -103,6 +102,22 @@ def get_projects():
     else:
         return make_response('Invalid parameters', 400)
 
+    try:
+        res = [x for x in res[:]]
+
+        for project in res:
+
+
```
Comment (henderkes) [inline backend/api/projects.py]: too many blank lines

File: backend/api/projects.py [hunk 5]
```diff
@@ -103,6 +102,22 @@ def get_projects():
...
+            project['is_bookmark'] = 'true' if str(project['_id'])\
+                in current_user['bookmarks'] else 'false'
+            project['is_owner'] = 'true' if current_user['email']\
+                in [author['email'] for author in project['authors']] else 'false'
+
+
```
Comment (henderkes) [inline backend/api/projects.py]: too many blank lines

File: backend/api/search.py [hunk 6]
```diff
@@ -47,7 +47,18 @@ def search_simple():
 
     try:
         res = g.es.search(index="knexdb", body=request_json)
-        return jsonify(res['hits'])
+        try:
+            for project in res:
```
Comment (henderkes) [inline backend/api/search.py]: shouldn't this be res['hits']?

File: backend/api/search.py [hunk 7]
```diff
@@ -154,7 +166,9 @@ def search_tag():
         }
     try:
         res = g.es.search(index="knexdb", body=request_json)
-        return jsonify(res['hits'])
+        res = make_response(jsonify(res['hits']))
```
Comment (henderkes) [inline backend/api/search.py]: is the make_response necessary? jsonify already returns a response object

File: backend/api/users.py [hunk 8]
```diff
@@ -172,4 +174,14 @@ def get_bookmarks():
...
+    try:
+        for project in res:
+            project['is_bookmark'] = 'true' if str(project['_id'])\
+                in current_user['bookmarks'] else 'false'
+            project['is_owner'] = 'true' if current_user['email']\
+                in [author['email'] for author in project['authors']]\
+                else 'false'
+
+    except KeyError:
+        pass
```
Comment (henderkes) [inline backend/api/users.py]: why is this in the /bookmarks call?!?

**Review by henderkes — DISMISSED** [review]

File: backend/api/search.py (hunk 7 above)
Comment (mharasic) [reply backend/api/search.py]: then its not neccessary... but headers should be set

File: backend/api/users.py (hunk 8 above)
Comment (mharasic) [reply backend/api/users.py]: everywhere a list of projects is taken, the meta-data has to be added... frontend-modules rely on this flag for visualisation

File: backend/api/search.py (hunk 7 above)
Comment (mharasic) [reply backend/api/search.py]: you are right... documentation already sets there header

File: backend/api/users.py (hunk 8 above)
Comment (henderkes) [reply backend/api/users.py]: but "res" is a user object, there are no "projects" and it's not list-iterable but dict-iterable, returning name, email, password, etc.

The call only returns a bookmark id

File: backend/api/search.py (hunk 6 above)
Comment (mharasic) [reply backend/api/search.py]: Yes

**Review by henderkes — COMMENTED** [review]
Functionality is tested right now, I did the changes I requested

File: backend/api/search.py [hunk 9]
```diff
@@ -27,20 +27,7 @@ def search_simple():
...
-                    'tags^2',
-                    'title^2',
-                    'description',
-                ],
-            },
-        },
-        'from': offset,
-        'size': count,
-    }
+    r
```
Comment (Drakulix) [inline backend/api/search.py]: r?

**Review by Drakulix — DISMISSED** [review]

File: backend/api/search.py (hunk 9 above)
Comment (henderkes) [reply backend/api/search.py]: working on it haha

**Review by henderkes — COMMENTED** [review]
waiting on search api tests from Frontend

**Comment by henderkes** [comment]
If search tests are running this is the fix for #274

**Review by Drakulix — APPROVED** [review]

**Pull request merged**